### PR TITLE
Tamu docs, Tamu CLI, test_cli.py, parallelize tests

### PR DESCRIPTION
**Pull request by ac6y**
Hi Dennis,
Added a few more commits - mostly docs for Tamu provider, tamu cli support and some tests:

* added dev-requirement pytest-xdist (so tests can run in parallel)
* updated Makefile to do same tests as Travis, and add "-n auto" to parallelize
* add cli param support for tamu (might need some of these for other providers too?)
* add test_cli.py module to test cli call functionality, (just tamu and default tests added initially)
* merged AUTHORS.rst and docs/authors.rst and replaced AUTHORS.rst with symlink to docs/authors.rst

Other stuff still ToDo:
* add cli param support if needed for other providers (assuming the way we're doing this is ok)
* add cli tests to test_cli.py for other providers

Cheers,
Michael

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ init:
 	pip install -r requirements.txt
 
 test:
-	py.test test_geocoder.py --verbose
+	py.test tests --doctest-modules --pep8 geocoder -v --cov geocoder --cov-report term-missing
```
Comment (DenisCarriere) [inline Makefile]: Glade you made this edit~! :+1:

**Comment by DenisCarriere** [comment]
Awesome work Michael! Great job, thanks for adding all the extra documentation for Tamu.

Cheers! :+1:

**Comment by ac6y** [comment]
Thank  you Denis! It's a pleasure to be able to contribute :)

Cheers,
Michael
:mobile:

**Comment by DenisCarriere** [comment]
The docs are updated on the Geocoder page.
https://geocoder.readthedocs.org/providers/Tamu.html

**Comment by DenisCarriere** [comment]
@ac6y Do you have a Twitter account? I couldn't find you. I posted this as a mention.
https://twitter.com/DenisCarriere/status/718173552474722304

**Comment by ac6y** [comment]
No, I've been meaning to get a Twitter account, but haven't yet. Thanks for
the mention tweet!

Cheers,
Michael
:mobile:

**Pull request merged**